docs/README.md

```diff
@@ -0,0 +1,62 @@
+These pages contain an introduction to Temporal's system architecture and internal implementation.
+
+Below we give a high-level overview of how Temporal works; other pages provide more detailed descriptions of the services and of the lifecycle of a Workflow Execution:
+
+- [History Service](./history-service.md)
+- [Matching Service](./matching-service.md)
+- [Temporal Workers](https://github.com/temporalio/sdk-core/blob/master/ARCHITECTURE.md)
+- [Workflow Lifecycle](./workflow-lifecycle.md)
```

Comment:
I would split "internal services" and "overall concepts". Especially in this list, it sounds like "Temporal Workers" section is about "service/worker" but it is not. Let's add a section with "Internal services" and list all 4 there: Frontend, History, Matching, Intreal worker. Link FE and IW to some "./TODO.md".  And then "Concepts" (or whatever section) with two other items.

Reply:
Thanks, changed to:

<table>
  <tr>
    <td><img width="245" alt="image" src="https://github.com/temporalio/temporal/assets/52205/53c411e6-2b13-429c-b17c-92962ba9f44f">
    </td>
  </tr
</table>